### ISSUE TITLE
Unable to run log2timeline.py due to issue with Python multiprocessing when running on MacOS in virtualenv

### ISSUE THREAD

**Issue by madsumm** [log2timeline/plaso#4578]
**Describe the problem:**

Please provide a clear and detailed description of what the problem is.

**To Reproduce:**

The version of Plaso you used:

For example: 20221229

The operating system you are running Plaso on (Not the operating system of the image/files you're trying to analyze):

For example: MacOS Monterey

Steps to reproduce the behavior including command line and arguments and output:

Java version 3.11.2 (Latest) for MacOS
Plaso installed in virtualenv, enabled python3.
ran "pip install -r requirements.txt" to ensure dependencies are updated in the virtualenv.
Ran "run_test.py" with no errors.

Ran "log2timeline.py --parsers win7_slow --hashers md5,sha256 --skip_compressed_streams --hasher_file_size_limit 20480000 --storage-file test.plaso test.E01"

**Debug output/tracebacks:**
```
Traceback (most recent call last):
  File "/Users/mathewsoon/plaso_env/bin/log2timeline.py", line 4, in <module>
    __import__('pkg_resources').run_script('plaso==20221229', 'log2timeline.py')
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 708, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1517, in run_script
    exec(code, namespace, namespace)
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/EGG-INFO/scripts/log2timeline.py", line 103, in <module>
    if not Main():
           ^^^^^^
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/EGG-INFO/scripts/log2timeline.py", line 77, in Main
    tool.ExtractEventsFromSources()
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/plaso/cli/extraction_tool.py", line 764, in ExtractEventsFromSources
    processing_status = self._ProcessSources(session, storage_writer)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/plaso/cli/extraction_tool.py", line 556, in _ProcessSources
    processing_status = extraction_engine.ProcessSources(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/plaso/multi_process/extraction_engine.py", line 996, in ProcessSources
    worker_process = self._StartWorkerProcess(process_name)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mathewsoon/plaso_env/lib/python3.11/site-packages/plaso-20221229-py3.11.egg/plaso/multi_process/extraction_engine.py", line 760, in _StartWorkerProcess
    process.start()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/spawn.py", line 187, in get_preparation_data
    main_mod_name = getattr(main_module.__spec__, "name", None)
                            ^^^^^^^^^^^^^^^^^^^^
AttributeError: module '__main__' has no attribute '__spec__'. Did you mean: '__doc__'?
```

Tried various versions.
Java 3.8.10.
Plaso various version. All have the same result.

**Comment by joachimmetz** [comment]
```
 File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/spawn.py", line 187, in get_preparation_data
    main_mod_name = getattr(main_module.__spec__, "name", None)
```

Is not an issue in the Plaso source 

> Java version 3.11.2 (Latest) for MacOS

Java ?

**Comment by madsumm** [comment]
Had moved JAVA from 3.11.2 to 3.8.10 (similar to SIFT). Same happen.
Setup on latest version of MacOS, same thing happen.
Move to previous version of Plaso, same thing happen. 
Will try similar version of Plaso in SIFT to test.

**Comment by madsumm** [comment]
Manage to get it running, by using ./tools/log2timeline.py instead of installed log2timeline.py

Hopefully this helps.

**Comment by joachimmetz** [comment]
Plaso does not use Java, and SIFT is not a supported installation. So I have no idea how you are trying to install it. Might be better to stick with the Docker installation. https://plaso.readthedocs.io/en/latest/sources/user/Users-Guide.html

> Manage to get it running, by using ./tools/log2timeline.py instead of installed log2timeline.py

what is `log2timeline.py` then? what creates it ? The source does not come with this file

**Comment by madsumm** [comment]
Hi, Sincere apologies if I had caused any confusion.

I did not install java or use java on my Mac. I use SIFT only to compare the versions of python and Plaso used to see if mine is affected. I was testing various versions of Python, not Java to see if that was the cause for the plaso install in mac.

For the last comment, I was referring to within download of your plaso-20230311.tar.gz, there is ./plaso-20230311/tools/log2timeline.py file which I used directly.

Example the following works 
`$ ./plaso-20230311/tools/log2timeline.py --storage-file test.plaso test.E01`

But the following does not work
`$ log2timeline.py --storage-file test.plaso test.E01`

I know docker works, but would have issue if I uses more than 8 CPUs. If possible I like to use your native plaso install directly using Virtualenv on my mac, which works the fastest for me before.
